### PR TITLE
57 user input api key

### PR DESCRIPTION
**Pull request by heatherlogan-scottlogic**

### CODE REVIEW

**Comment by heatherlogan-scottlogic** [comment]
Requires ts-frontend to be merged

File: frontend/src/components/ApiKeyBox/ApiKeyBox.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        id="api-key-input"
+        className={`api-key-input ${isValidated ? "validated" : ""} ${
+          isInvalidated ? "invalidated" : ""
+        }`}
+        type="text"
+        value={apiKey}
+        placeholder="enter your API key here.."
+        onChange={handleApiKeyChange}
+        onKeyUp={handleApiKeySubmit.bind(this)}
+      />
```
Comment (gsproston-scottlogic) [inline frontend/src/components/ApiKeyBox/ApiKeyBox.tsx:63]: Would it be possible to hide the key in some way? I'm thinking like a password input. Just so the user doesn't have their API key on display all the time.

File: backend/src/openai.js [hunk 2]
```diff
@@ -36,29 +36,61 @@ const chatGptFunctions = [
...
+}
+
+function initOpenAi(session) {
   configuration = new Configuration({
-    apiKey: process.env.OPENAI_API_KEY,
+    apiKey: session.apiKey,
   });
   openai = new OpenAIApi(configuration);
+  console.debug("OpenAI initialised");
 }
```
Comment (gsproston-scottlogic) [inline backend/src/openai.js:94]: Could we have it so if the .env contains an OPENAI_API_KEY, then it uses that by default, and doesn't require the user to input a key. Would be easier for dev work.

File: frontend/src/service/openaiService.ts
Comment (gsproston-scottlogic) [inline frontend/src/service/openaiService.ts:1]: This file got renamed to 'chatService.ts' so it looks like GitHub thinks this is a new file.

**Review by gsproston-scottlogic — CHANGES_REQUESTED** [review]

**Review by gsproston-scottlogic — APPROVED** [review]
Looks great!

**Pull request merged**